src/codegen_any.cpp

```
@@ -32,9 +32,7 @@ bool CodeGen::isNonStrPointer(llvm::Value *val) {
     }
 
     // Function pointers
-    if (fn_type_info_.count(val)) return true;
-    if (auto *load = llvm::dyn_cast<llvm::LoadInst>(val))
-        if (fn_type_info_.count(load->getPointerOperand())) return true;
+    if (lookupFnTypeInfo(val) != fn_type_info_.end()) return true;
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)